utils/build-script

```diff
@@ -1450,7 +1481,17 @@ details of the setups of other systems or automated environments.""")
     if args.dry_run:
         build_script_impl_args += ["--dry-run"]
 
-    call_without_sleeping([build_script_impl] + build_script_impl_args)
+    # Compute the set of host-specific variables, which we pass through to the
+    # build script via environment variables.
+    host_specific_variables = compute_host_specific_variables(args)
+    build_script_impl_env = {}
+    for (host, options) in host_specific_variables.items():
+        for (name, value) in options.items():
+            build_script_impl_env["HOST_VARIABLE_{}__{}".format(
+                host, name)] = value
+        
+    call_without_sleeping([build_script_impl] + build_script_impl_args,
+                          env=build_script_impl_env)
```

Comment:
Does this function print the environment in case the command fails?  (It seems like it doesn't.)  This is critical for reading logs from various automated systems like CI.